src/java/main/org/apache/zookeeper/server/quorum/Learner.java

```diff
@@ -253,7 +255,10 @@ protected void connectToLeader(InetSocketAddress addr)
                     throw new IOException("initLimit exceeded on retries.");
                 }
 
-                sockConnect(sock, addr, Math.min(self.tickTime * self.syncLimit, remainingInitLimitTime));
+                sockConnect(sock, leader.addr, Math.min(self.tickTime * self.syncLimit, remainingInitLimitTime));
```

Comment:
leader could be null in leader.addr right? Still wondering why change the leader type from an address to a QuorumServer.